maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/main/java/org/apache/maven/scm/provider/git/jgit/command/JGitUtils.java

```diff
@@ -576,8 +576,8 @@ public static List<ScmFile> removeAllFiles( Git git, ScmFileSet fileSet )
             // really tracked
             for ( Iterator<File> itfl = fileSet.getFileList().iterator(); itfl.hasNext(); )
             {
-                String path = FilenameUtils.normalizeFilename( relativize( baseUri, itfl.next() ) );
-                if ( path.equals( FilenameUtils.normalizeFilename( scmFile.getPath() ) ) )
+                String path = relativize( baseUri, itfl.next() );
+                if ( path.equals( scmFile.getPath() ) )
```

Comment:
this comparison is still wrong, IMHO you first need to make `itfl.next()` absolute and then relativize against `workingCopyRootUri`.

Reply:
Correct. Please fork and provide a PR on top. You know better what to do already.